rts/motoko-rts/src/region.rs

```diff
@@ -671,19 +671,23 @@ pub(crate) unsafe fn region_migration_from_some_stable_memory<M: Memory>(mem: &m
      * Initialize region0's table entries, for "recovery".
      */
 
-    /* head block is physically last, but logically first. */
-    let head_block_id = region0_blocks - 1; /* invariant: region0_blocks is non-zero. */
-
-    meta_data::block_region_table::set(BlockId(head_block_id as u16), Some((RegionId(0), 0)));
-    meta_data::region_table::set(&RegionId(0), Some(RegionSizeInPages(region0_pages as u64)));
+    /* head block is physically last block, but logically first block in region. */
+    let head_block_id: u16 = (region0_blocks - 1) as u16; /* invariant: region0_blocks is non-zero. */
```

Comment:
Idea (not very important): Maybe a check that `region0_blocks` fits into u16 would make the downcast look safer. This relies on the assumption that `max::BLOCKS` is never changed to exceed `u16::MAX`.

Reply:
Added the assert